php/PaypalIPN.php

```diff
@@ -136,11 +136,10 @@ public function verifyIPN()
 
         curl_close($ch);
 
-        // Check if PayPal verifies the IPN data, and if so, return true.
+        // Check if PayPal verifies the IPN data, and if so, return data array.
         if ($res == self::VALID) {
-            return true;
-        } else {
-            return false;
+            return $myPost;
```

Comment:
I think this class should have a single responsibility, to verify that the IPN data is valid.
You can then retrieve the post data from `$__POST` or your frameworks request object. I do not see a reason why this needs to return data.